src/main/java/pl/creative/rental_server/core/features/moderator/item/ItemService.java

```diff
@@ -42,27 +43,17 @@ public class ItemService {
 
     @Transactional
     public GetItemDto addItem(FillItemDto fillItemDto) {
-        log.info("Adding new item");
+        log.info("Adding new item - {}",fillItemDto.getName());
         Item itemToSave = itemMapper.mapFillItemDtoToItem(fillItemDto);
         String uuid = randomIdHandler.generateUniqueIdFromTable(itemRepository);
         itemToSave.setId(uuid);
         itemToSave.setDateOfCreation(LocalDateTime.now());
         for (String categoryId : fillItemDto.getCategoriesId()) {
             if (categoryId != null) {
-                categoryRepository.findById(categoryId).ifPresentOrElse(itemToSave::addCategoryToCategoryIds,
-                        () -> {
-                            log.error("Category with that id {} does not exist");
-                            throw new CategoryNotFound(String.format("Category with that id %s does not exist", categoryId));
-                        }
-                );
+                addCategoryToItem(itemToSave,categoryId);
             }
```

Comment:
What if category with such id does not exist?

Reply:
Please read with understanding -> i only extracted method not changed any logic inside . It will check and assign or throw when id not exists